src/os/bluestore/BlueStore.h

```diff
@@ -244,6 +244,7 @@ class BlueStore : public ObjectStore,
     }
 
     bool empty() const {
+      std::lock_guard<std::mutex> l(cache->lock);
       return buffer_map.empty();
     }
```

Comment:
This isn't right--we deadlock with this patch.  (Please run ./ceph_test_objectstore --gtest_filter=*/2 for any bluestore changes!).  The cache->lock protects LRU state, not the forward mapping structures in BufferSpace and OnodeSpace.  Please drop this patch!